repo/repo.go

```diff
@@ -89,17 +99,54 @@ func ListRepositories(workspace *rule.File) (repos []*rule.Rule, repoFileMap map
 			if err != nil {
 				return nil, nil, err
 			}
+			directiveRepoMap, err = parseRepositoryDirectives(macroFile.Directives)
+			if err != nil {
+				return nil, nil, err
+			}
+			for _, repo := range directiveRepoMap {
+				repos = append(repos, repo)
+				repoFileMap[repo.Name()] = macroFile
+			}
 			for _, repo := range macroFile.Rules {
 				if name := repo.Name(); name != "" {
-					repos = append(repos, repo)
-					repoFileMap[name] = macroFile
+					if _, ok := directiveRepoMap[name]; !ok {
+						repos = append(repos, repo)
+						repoFileMap[name] = macroFile
+					}
 				}
 			}
 		}
 	}
 	return repos, repoFileMap, nil
 }
 
+func parseRepositoryDirectives(directives []rule.Directive) (repoMap map[string]*rule.Rule, err error) {
+	repoMap = make(map[string]*rule.Rule)
+	for _, d := range directives {
+		switch d.Key {
+		case "repository":
+			vals := strings.Split(d.Value, " ")
+			if len(vals) < 2 {
+				return nil, fmt.Errorf("Failure parsing repository: %s, you must define at least a rule kind and name attribute", d.Value)
+			}
+			kind := vals[0]
+			r := rule.NewRule(kind, "")
+			for _, val := range vals[1:] {
+				kv := strings.SplitN(val, "=", 2)
+				if len(kv) != 2 {
+					return nil, fmt.Errorf("Failure parsing repository: %s, expected format for attributes is <attr1_name=attr1_value>", d.Value)
```

Comment:
Nit: drop the `<` and `>` here. It would be easy to misinterpret that, thinking those characters are needed literally.